src/gui/owncloudsetupwizard.cpp

```diff
@@ -610,32 +638,62 @@ void OwncloudSetupWizard::slotAssistantFinished(int result)
-                        QStringList() << QLatin1String("/"));
+            if (!hasServerConfig) {
+                qCInfo(lcWizard) << "Adding folder definition for" << localFolder << _remoteFolder;
+                FolderDefinition folderDefinition;
+                folderDefinition.localPath = localFolder;
+                folderDefinition.targetPath = FolderDefinition::prepareTargetPath(_remoteFolder);
+                folderDefinition.ignoreHiddenFiles = folderMan->ignoreHiddenFiles();
+                if (_ocWizard->useVirtualFileSync()) {
+                    folderDefinition.virtualFilesMode = bestAvailableVfsMode();
+                }
+                if (folderMan->navigationPaneHelper().showInExplorerNavigationPane())
+                    folderDefinition.navigationPaneClsid = QUuid::createUuid();
+
+                auto f = folderMan->addFolder(account, folderDefinition);
+                if (f) {
+                    if (folderDefinition.virtualFilesMode != Vfs::Off && _ocWizard->useVirtualFileSync())
+                        f->setNewFilesAreVirtual(true);
+
+                    f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncBlackList,
+                        _ocWizard->selectiveSyncBlacklist());
+                    if (!_ocWizard->isConfirmBigFolderChecked()) {
+                        // The user already accepted the selective sync dialog. everything is in the white list
+                        f->journalDb()->setSelectiveSyncList(SyncJournalDb::SelectiveSyncWhiteList,
+                            QStringList() << QLatin1String("/"));
+                    }
+                }
+            } else {
+                auto array = configJson.object()["default_folders"].toArray();
+                // FIXME!  check consistency of the config and that the folder can be created
+                for (const auto &x : array) {
+                    QString remoteFolder = x.toObject()["remote_folder"].toString();
+                    QString name = QFileInfo(remoteFolder).fileName();
+                    FolderDefinition folderDefinition;
+                    folderDefinition.localPath = (array.size() == 1) ? localFolder : QString(localFolder + name + '/');
+                    QDir(localFolder).mkpath(folderDefinition.localPath);
+                    folderDefinition.targetPath = FolderDefinition::prepareTargetPath(remoteFolder);
+                    folderDefinition.ignoreHiddenFiles = folderMan->ignoreHiddenFiles();
+                    if (folderMan->navigationPaneHelper().showInExplorerNavigationPane())
+                        folderDefinition.navigationPaneClsid = QUuid::createUuid();
+
+                    auto f = folderMan->addFolder(account, folderDefinition);
+                    if (f) {
+                        if (folderDefinition.virtualFilesMode != Vfs::Off && _ocWizard->useVirtualFileSync())
+                            f->setNewFilesAreVirtual(true);
```

Comment:
Can be removed I think. folderDefinition.virtualFilesMode is never set and useVirtualFileSync() an't be true.